### DISCUSSION TITLE
Custom languages

### DISCUSSION THREAD

**Discussion by shishisana** [optimajet/formengine#14] [Q&A]
How to implement a custom language with static HTML, currently using https://github.com/optimajet/workflowengine.net/tree/master/designer

**Comment by optimajet** [comment]
@shishisana Do you need a custom language for Workflow Engine Designer or FormEngine?

**Reply by jkagot** [reply]
Hi, how to add custom language for FormEngine?

**Reply by optimajet** [reply]
You can use the i18n property to pass custom languages (https://formengine.io/documentation/api-reference/interfaces/react_form_builder_designer.FormBuilderProps#i18n).

Code example:
```typescript
import {frFR, trTR} from 'rsuite/esm/locales/index.js'

const builderLanguages: FormBuilderI18n = {
  languages: [
    new Language('fr', 'FR', 'French', 'Français'),
    new Language('tr', 'TR', 'Turkish', 'Türkçe'),
  ],
  getData: async languageFullCode => {
    if (languageFullCode === 'tr-TR') {
      return {
        data: {leftMenu: {menu: 'Menü'}},
        componentsLocale: trTR
      }
    }
    return {
      data: {leftMenu: {menu: 'Menu'}},
      componentsLocale: frFR
    }
  }
}

<FormBuilder 
   ...
   i18n={builderLanguages}
/>
```


See also this issue - https://github.com/optimajet/formengine/issues/13.
